tools/scalar-schema/src/scalar_schema/jdbc.clj

```diff
@@ -0,0 +1,298 @@
+       (if index-order (str "'" index-order "'") "NULL") ","
+       ordinal-position
+       ")"))
+
+(defn- get-key-type
+  [column partition-key clustering-key]
+  (cond
+    (key? column partition-key) "PARTITION"
+    (key? column clustering-key) "CLUSTERING"
+    :else nil))
+
+(defn- get-secondary-indexed [rdb-engine column secondary-index]
+  (if (key? column secondary-index) (boolean-value-true rdb-engine)
+      (boolean-value-false rdb-engine)))
+
+(defn- zipseq
+  ([coll] (partition 1 coll))
+  ([c1 & colls] (let [c (cons c1 colls)] (partition (count c) (apply interleave c)))))
+
+(defn- insert-metadata
+  [db-spec rdb-engine prefix schema table partition-key clustering-key columns secondary-index]
+  (doall (map #(let [column (first (first %))
+                     data-type (string/upper-case (second (first %)))
+                     key-type (get-key-type column partition-key clustering-key)
+                     key-order (if (key? column clustering-key) "ASC" nil)
+                     secondary-indexed (get-secondary-indexed rdb-engine column secondary-index)
+                     index-order (if (key? column secondary-index) "ASC" nil)
+                     ordinal-position (second %)]
+                 (execute! db-spec (make-insert-metadata-statement rdb-engine prefix schema table
+                                                                   column data-type key-type
+                                                                   key-order secondary-indexed
+                                                                   index-order ordinal-position)))
+              (zipseq columns (iterate inc 1)))))
+
+(defn- make-drop-table-statement
+  [rdb-engine schema table]
+  (str "DROP TABLE " (get-table-name rdb-engine schema table)))
+
+(defn- drop-table
+  [db-spec rdb-engine schema table]
+  (execute! db-spec (make-drop-table-statement rdb-engine schema table)))
+
+(defn- make-delete-metadata-statement
+  [rdb-engine prefix schema table]
+  (let [full_table_name (get-table-name schema table)]
+    (str "DELETE FROM "
+         (get-metadata-table-name rdb-engine prefix)
+         " WHERE " (enclose rdb-engine "full_table_name") " = "
+         "'" full_table_name "'")))
+
+(defn- delete-metadata
+  [db-spec rdb-engine prefix schema table]
+  (execute! db-spec (make-delete-metadata-statement rdb-engine prefix schema table)))
+
+(defn- reorder-columns
+  [columns partition-key clustering-key]
+  (concat (map #(vector % (get columns %)) partition-key)
+          (map #(vector % (get columns %)) clustering-key)
+          (filter #(not (key? (first %) (concat partition-key clustering-key)))
+                  (apply list columns))))
+
+(defn make-jdbc-operator
+  [{:keys [jdbc-url user password prefix]}]
+  (let [db-spec (get-db-spec jdbc-url user password)
+        rdb-engine (get-rdb-engine jdbc-url)]
+    (create-metadata-schema db-spec rdb-engine prefix)
+    (create-metadata-table db-spec rdb-engine prefix)
```

Comment:
In the design, I think this function `make-jdbc-operator` should just only make the operator.
And, the internal functions are not always evaluated immediately for lazy evaluation.